dept44-example/src/main/java/se/sundsvall/petinventory/integration/db/model/PetImageEntity.java

```diff
@@ -0,0 +1,174 @@
+package se.sundsvall.petinventory.integration.db.model;
+
+import static org.hibernate.Length.LONG32;
+
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.Objects;
+
+import org.hibernate.annotations.OnDelete;
+import org.hibernate.annotations.OnDeleteAction;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EntityListeners;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.ForeignKey;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import se.sundsvall.petinventory.integration.db.model.listener.PetImageEntityListener;
+
+@Entity
+@Table(name = "pet_image")
+@EntityListeners(PetImageEntityListener.class)
+public class PetImageEntity {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id")
+	private Long id;
+
+	@Column(name = "file_name")
+	private String fileName;
+
+	@Column(name = "mime_type")
+	private String mimeType;
+
+	@Column(name = "content", length = LONG32)
+	private byte[] content;
+
+	@Column(name = "created")
+	private OffsetDateTime created;
+
+	@Column(name = "modified")
+	private OffsetDateTime modified;
+
+	@ManyToOne(optional = false, fetch = FetchType.EAGER)
+	@JoinColumn(name = "pet_name_id", nullable = false, updatable = false, foreignKey = @ForeignKey(name = "fk_pet_image_pet_name_id_pet_name_id"))
+	@OnDelete(action = OnDeleteAction.CASCADE)
+	private PetNameEntity petName;
+
+	public static PetImageEntity create() {
```

Comment:
No big deal, but maybe add a private constructor if we're only supposed to use this method?

Reply:
This is not possible since this would violate the JPA-specification:

https://openjpa.apache.org/builds/1.2.3/apache-openjpa/docs/jpa_overview_pc.html#:~:text=The%20JPA%20specification%20requires%20that,include%20a%20no%2Darg%20constructor.

Reply:
Didn't think about that =D